### PR TITLE
[ERK-216] Google Tag Manager

### PR DESCRIPTION
**Pull request by artismarti**
This PR adds the following tracking to the InfoBox in Atomic Examples:
* Github link
* Open new window link
* View all examples link

### CODE REVIEW

File: src/components/MessageReactions/MessageReactions.tsx [hunk 1]
```diff
@@ -18,7 +22,12 @@ const MessageReactions = () => {
   }
 
   useEffect(() => {
-    sendMessage()
+    sendMessage(),
+      setProjectInfo({
+        name: 'Message Reactions',
+        repoNameAndPath:
+          'atomic-examples/tree/main/src/components/MessageReactions',
+      })
```
Comment (dpiatek) [inline src/components/MessageReactions/MessageReactions.tsx:30]: It looks like indentation is off here along with missing semicolons?

File: src/Layout.tsx [hunk 2]
```diff
@@ -32,7 +44,12 @@ const Layout = () => {
     <main className="h-screen flex pt-6 md:pt-0 md:items-center justify-center font-sans">
       <Outlet context={{ channelName: channelId, clientId, setProjectInfo }} />
       <div className="fixed bottom-0 md:absolute md:left-12 md:bottom-12">
-        <InfoCard projectInfo={projectInfo} />
+        <InfoCard
+          projectInfo={projectInfo}
+          openNewTabTrackerEvent={() => openNewWindow(projectInfo.name)}
+          githubRepoLinkTrackerEvent={() => viewRepoAtGithub(projectInfo.name)}
+          allExamplesLinkTrackerEvent={() => viewAllExamples(projectInfo.name)}
```
Comment (dpiatek) [inline src/Layout.tsx:51]: Maybe the names of these props should be closer to the names of the functions (`openNewTabTrackerEvent` -> `openNewWindow`, why not `openNewTabTrackerEvent` -> `openNewTabTrackerEvent`)? They seem to be saying the same thing, but in different ways.

File: src/Layout.tsx [hunk 3]
```diff
@@ -3,9 +3,21 @@ import { Outlet, useSearchParams } from 'react-router-dom'
...
 import { configureAbly } from '@ably-labs/react-hooks'
 import InfoCard from './InfoCard'
+import { viewRepoAtGithub, viewAllExamples, openNewWindow } from './gtm'
 
 const clientId = nanoid()
 
+import TagManager from 'react-gtm-module'
+
+const tagManagerArgs = {
+  gtmId: 'GTM-TZ37KKW',
```
Comment (dpiatek) [inline src/Layout.tsx:13]: ideally we could pass this through an ENV variable, but that can be refactored too

**Review by dpiatek — CHANGES_REQUESTED** [review]

File: src/Layout.tsx (hunk 2 above)
Comment (artismarti) [reply src/Layout.tsx:51]: that makes sense. @dpiatek 
I was also considering having just one function & passing the event to it. Or would you say it's better to leave it explicit as different functions like this?

**Comment by artismarti** [comment]
Closing this PR since we decided to use UTM tags instead.

**Comment by artismarti** [comment]
https://github.com/ably-labs/atomic-examples/pull/22

**Pull request closed without merging**